types/aws-lambda/trigger/msk.d.ts

```diff
@@ -2,6 +2,10 @@ import { Handler } from '../handler';
 
 export type MSKHandler = Handler<MSKEvent, void>;
 
+export interface MSKRecordHeader {
+    [headerKey: string]: Uint8Array;
```

Comment:
You won't get a Uint8Array - the event value is the result of `JSON.parse`, so unfortunately this should be `number[]`.

The difference matters for members like `buffer` and `set()`

Reply:
Verified.
```typescript
```

```
headers type [object Array]
```
I'm curious as to where this is converted from the binary wire format of Kafka into a JSON string.  That would imply a somewhat inefficient/extra serialization/deserialization.   Is the record converted into a CloudWatch event as part of the trigger and that's where the JSON serialization is happening?

Reply:
To my understanding, Lambda can only take JSON: or at least in the provided runtimes. So it would likely be in the trigger code, yes.